docs/software_requirements/queues.sdoc

```diff
@@ -0,0 +1,216 @@
+[DOCUMENT]
+TITLE: Tracing
+REQ_PREFIX: ZEP-SRS-20-
+
+[GRAMMAR]
+IMPORT_FROM_FILE: software_requirements.sgra
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-1
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue definition at compile time
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to define and initialize a queue at compile time.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-2
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue definition at run time.
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to define and initialize a queue at run time.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-3
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue append item to the end
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to append a data item to the end of a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-4
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue append item to the start
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to append a data item in the first place of a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-5
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue remove item
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to remove a data item from a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-6
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue get item
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to get a data item from a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-7
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue get item timeout
+STATEMENT: >>>
+When a timout for the get mechanism to get a data item is configured, the zephyr RTOS shall wait
+that period to optain the data item from the queue.
```

Comment:
"... the get mechanism to get a data item... " seems rather complicated
"...is configured..." to me sounds like I configure once and then it works that way rather than like a parameter.

Simpler suggestion, but incomplete behavior description:

```suggestion
The queues get mechanism shall provide a timeout parameter, for Zephyr RTOS to wait for
that time period to obtain the data item from the queue.
```
Rather complete behavior but again complex. Otherwise need more requirements:

```suggestion
The queues get mechanism shall provide a timeout parameter, for Zephyr RTOS to maximum wait for
that time period to obtain the data item from the queue, if none is available, before, if unsuccessful, returning with an error.
```